drivers/unix/os_unix.cpp

```diff
@@ -582,6 +584,9 @@ Dictionary OS_Unix::execute_with_pipe(const String &p_path, const List<String> &
 	err_pipe.instantiate();
 	err_pipe->open_existing(pipe_err[0], 0);
 
+	ProcessInfo pi;
+	process_map->insert(pid, pi);
```

Comment:
I copied code from Windows implementation.

Reply:
> Also if there is nothing to clean up data from this HashMap, will it leak?

Same as Windows, it's cleaned on exit when hashmap is deleted. The amount of used memory should be negligible in most use cases, but if it still is a concern, it can be changed from `HashMap<>` to `LRU<>` with some reasonably high capacity.

> My personal preference would be to keep the old design and document that you have to pick one or the other function. Use is_process_running only if you do not care about the exit code.

This would be fine if it was the same on Windows, but it's not. I do not think it's adding too much complexity, but it's keeping it consistent. And adding a single mutex should be enough to make it thread safe.

Reply:
A mutex would be enough to satisfy my thread safety concerns.

If you take a game server written in godot that executes processes once a second and runs for weeks, even then the hashmap will get to millions of elements but probably not enough to cause a huge memory problem. I am not going to make a fuss over this.

Reply:
I added a Mutex (though I'm not familiar with mutexes in core, so not sure if correctly).